### PR TITLE
Obsolesce the Voider

### PR DESCRIPTION
**Pull request by DanielWright**
(This PR has been resubmitted against #11, because of an exceedingly bizarre Travis bug.)

This PR ended up getting a bit out of hand. The original aim was to obsolesce the Voider service object. Unfortunately, this revealed the absence of a request spec for voiding a transaction on Shopify. This request spec, once written, revealed issues with the implementation.

After some wrangling, the voiding workflow is implemented and covered with tests. Some refactoring remains, but nothing worth delaying a commit over.

### CODE REVIEW

**Comment by DanielWright** [comment]
I think this is ready for CR.

**Comment by DanielWright** [comment]
I'm not sure why this PR isn't being picked up my Travis. Maybe because the merge isn't proposed to master? Anyway, the full suite passes:

File: app/services/spree/retail/shopify/refunder.rb [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
-        @refund_reason = options[:reason]
-        @order_id = options[:order_id]
+      def initialize(credited_money:, transaction:, refunder_interface: nil, can_issue_refund_policy_klass: nil, **options)
         @credited_money = BigDecimal.new(credited_money)
-        @transaction_interface = transaction_interface || default_transaction_interface
+        @transaction = transaction
+        @refund_reason = options[:reason]
+
+        # FIXME: HOOOOOOORJ This is abominable
+        @order_id = transaction.prefix_options[:order_id]
```
Comment (DanielWright) [inline app/services/spree/retail/shopify/refunder.rb]: This is the only really gnarly bit of implementation here. `transaction` _claims_ it responds to `.order_id`, but its response is always nil. I'm tempted to monkeypatch `ShopifyAPI::Transaction` to fix its interface. Thoughts?

Comment (DanielWright) [reply app/services/spree/retail/shopify/refunder.rb]: I think a monkeypatch is acceptable. There are [numerous resource](https://github.com/Shopify/shopify_api/search?utf8=%E2%9C%93&q=prefix_options) that implement an explicit `order_id` method. I think the `init_prefix` macro is broken and/or is bad.

**Comment by DanielWright** [comment]
Either of you guys available for a review here?

**Comment by thisiscab** [comment]
LGTM! 👍

File: app/models/spree/gateway/shopify_gateway.rb [hunk 2]
```diff
@@ -20,8 +20,9 @@ def credit(money, transaction_id, gateway_options)
       provider.refund(money, transaction_id, options)
     end
 
-    def void(transaction_id, gateway_options)
-      pos_order_id = gateway_options[:originator].pos_order_id
+    def void(transaction_id, _)
+      pos_order_id = payments.find_by(response_code: transaction_id).pos_order_id
```
Comment (DanielWright) [inline app/models/spree/gateway/shopify_gateway.rb]: Are we concerned about the LoD violation here?

**Pull request merged**